src/Memory/MemoryUtils.cpp

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "MemoryUtils.h"
+
+#include <absl/strings/str_split.h>
+#include <stdlib.h>
+
+#include <string>
+
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/ReadFileToString.h"
+#include "OrbitBase/Result.h"
+
+using orbit_grpc_protos::TotalMemoryUsageInfo;
+
+std::optional<TotalMemoryUsageInfo> ParseMemInfo(const std::string& meminfo_content) {
+  if (meminfo_content.empty()) return std::nullopt;
+
+  std::vector<std::string> lines = absl::StrSplit(meminfo_content, '\n');
+  if (lines.empty()) return std::nullopt;
+
+  constexpr int kNumLines = 5;
+  constexpr int kBase = 10;
+  constexpr int kMissingInfo = -1;
+  int mem_total = kMissingInfo, mem_free = kMissingInfo, mem_available = kMissingInfo,
+      mem_buffers = kMissingInfo, mem_cached = kMissingInfo;
+  for (int i = 0, end = lines.size() > kNumLines ? kNumLines : lines.size(); i < end; ++i) {
+    const std::string& current_line = lines[i];
+    std::vector<std::string> splits = absl::StrSplit(current_line, ' ', absl::SkipWhitespace{});
+    if (splits.size() < 2) continue;
+    if (splits[0] == "MemTotal:") {
+      mem_total = std::strtoll(splits[1].c_str(), nullptr, kBase);
```

Comment:
absl::SimpleAtoi?

Reply:
Changed to use `absl::SimpleAtoi`. Thanks for telling me about this method.